### ISSUE TITLE
Open to publishing on brew?

### ISSUE THREAD

**Issue by chmac** [rentzsch/stressdrive#16]
Firstly, thanks for stressdrive, it's really useful to be able to test a whole hard drive.

I saw the install instructions of `port install stressdrive`. I already have brew installed, and I see `brew install ...` as one of the recommendations to get openssl. I wonder if you're up for publishing to brew?

I have helped a few other projects publish to brew, and so I know a little about the process. I'd be happy to lend a hand if that's helpful.

**Comment by rentzsch** [comment]
Yup, it would be nice for stressdrive to be on Homebrew. I have [mogenerator](https://github.com/rentzsch/mogenerator) on homebrew.

The main issue is Homebrew admirably wants to compile from source, and stressdrive's use of OpenSSL complicates the build.

Ideally stressdrive abandons OpenSSL since I just use it for SHA-1, and use its own internal implementation.

I think I looked at this a couple years back hoping to find a simple .h+.c implementation that uses processor intrinsics when compile-time-available (ideally ideally runtime-available). I found a project that was 90% there, but wasn't complete. I need to re-find that project and see if it's done or maybe I complete it for stressdrive.

**Comment by rentzsch** [comment]
Correction: since v1.2 we also use OpenSSL for AES as a fast in-process [CPRNG](https://en.wikipedia.org/wiki/Cryptographically_secure_pseudorandom_number_generator). So that's two things we use OpenSSL for, but are somewhat easy to naively integrate and drop OpenSSL interop woes.

For the record, https://github.com/noloader/SHA-Intrinsics was the project I was thinking of above. It's good documentation, but I have something more dynamic in mind.

**Comment by toy** [comment]
@chmac There are [a lot of formulaes](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-core%20depends_on%20%22openssl%403%22&type=code) that depend on openssl, for example [formula for http](https://github.com/Homebrew/homebrew-core/blob/master/Formula/h/httpd.rb) can be used as an example
